src/tests/vendor_agnostic/fixtures.py

```diff
@@ -0,0 +1,385 @@
+    # Load pre-defined datasets
+    load_generic_testing_data(engine, test_table_name_dict)
+    yield engine
+    tear_down_generic_testing_data(engine, test_table_name_dict)
+    engine.dispose()
+
+
+@pytest.fixture
+def ip_with_SQLite(ip_empty, setup_SQLite):
+    configKey = "SQLite"
+    alias = _testing.DatabaseConfigHelper.get_database_config(configKey)["alias"]
+
+    # Select database engine, use different sqlite database endpoint
+    ip_empty.run_cell(
+        "%sql "
+        + _testing.DatabaseConfigHelper.get_database_url(configKey)
+        + " --alias "
+        + alias
+    )
+    yield ip_empty
+    # Disconnect database
+    ip_empty.run_cell("%sql -x " + alias)
+
+
+@pytest.fixture(scope="session")
+def setup_duckDB(test_table_name_dict, skip_on_live_mode):
+    engine = create_engine(_testing.DatabaseConfigHelper.get_database_url("duckDB"))
+    # Load pre-defined datasets
+    load_generic_testing_data(engine, test_table_name_dict)
+    yield engine
+    tear_down_generic_testing_data(engine, test_table_name_dict)
+    engine.dispose()
+
+
+@pytest.fixture
+def ip_with_duckDB(ip_empty, setup_duckDB):
+    configKey = "duckDB"
+    alias = _testing.DatabaseConfigHelper.get_database_config(configKey)["alias"]
+
+    # Select database engine, use different sqlite database endpoint
+    ip_empty.run_cell(
+        "%sql "
+        + _testing.DatabaseConfigHelper.get_database_url(configKey)
+        + " --alias "
+        + alias
+    )
+    yield ip_empty
+    # Disconnect database
+    ip_empty.run_cell("%sql -x " + alias)
+
+
+@pytest.fixture(scope="session")
+def setup_MSSQL(test_table_name_dict, skip_on_live_mode):
+    with _testing.mssql():
+        engine = create_engine(_testing.DatabaseConfigHelper.get_database_url("MSSQL"))
+        # Load pre-defined datasets
+        load_generic_testing_data(engine, test_table_name_dict)
+        yield engine
+        tear_down_generic_testing_data(engine, test_table_name_dict)
+        engine.dispose()
+
+
+@pytest.fixture
+def ip_with_MSSQL(ip_empty, setup_MSSQL):
+    configKey = "MSSQL"
+    alias = _testing.DatabaseConfigHelper.get_database_config(configKey)["alias"]
+
+    # Select database engine
+    ip_empty.run_cell(
+        "%sql "
+        + _testing.DatabaseConfigHelper.get_database_url(configKey)
+        + " --alias "
+        + alias
+    )
+    yield ip_empty
+    # Disconnect database
+    ip_empty.run_cell("%sql -x " + alias)
+
+
+@pytest.fixture(scope="session")
+def setup_Snowflake(test_table_name_dict, skip_on_local_mode):
+    engine = create_engine(_testing.DatabaseConfigHelper.get_database_url("Snowflake"))
+    engine.connect()
+    # Load pre-defined datasets
+    load_generic_testing_data(engine, test_table_name_dict, index=False)
+    yield engine
+    tear_down_generic_testing_data(engine, test_table_name_dict)
+    engine.dispose()
+
+
+@pytest.fixture
+def ip_with_Snowflake(ip_empty, setup_Snowflake, pytestconfig):
```

Comment:
tony already created fixtures for all the databases, can't we use the same declarations, seems like we're duplicating a lot of code here

Reply:
Yeah I am thinking probably current `src/tests/vendor_agnostic` folder can be part of `src/tests/integration`